src/views/collection-view/index.tsx

```diff
@@ -1,29 +1,108 @@
 import Link from 'next/link'
+import { TutorialLite as ClientTutorialLite } from 'lib/learn-client/types'
+import useCurrentPath from 'hooks/use-current-path'
+import { useCurrentProduct } from 'contexts'
 import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
 import CoreDevDotLayout from 'layouts/core-dev-dot-layout'
-import { TutorialLite as ClientTutorialLite } from 'lib/learn-client/types'
-import ProductCollectionsSidebar from 'components/tutorials-sidebar/compositions/product-collections-sidebar'
-import { getTutorialSlug } from './helpers'
+import {
+  generateProductLandingSidebarNavData,
+  generateTopLevelSidebarNavData,
+} from 'components/sidebar/helpers'
+import TutorialsSidebar, {
+  HorizontalRule,
+  SectionList,
+  SectionTitle,
+} from 'components/tutorials-sidebar'
+import { CollectionCategorySidebarSection, getTutorialSlug } from './helpers'
 import { CollectionPageProps } from './server'
 import CollectionMeta from './components/collection-meta'
 
 function CollectionView({
   collection,
-  product,
   layoutProps,
 }: CollectionPageProps): React.ReactElement {
-  const { name, slug, description, shortName, tutorials } = collection
+  const currentProduct = useCurrentProduct()
```

Comment:
Isn't `product` passed anyways for the other layout pieces? Any reason why we wouldn't just use the prop here instead of the hook?

Reply:
Either way would work! I don't have a strong preference for either. I just used the hook here so the API for `CollectionView` didn't _have_ to receive the `product` object. It's mainly passed through  for `App` to hydrate the context. It's just coincidental that we can grab the prop here, not necessarily standard for every view component to be able to do it.

Reply:
Since the data is already being passed, imo it would make more sense to just use that! It's up to you though 👍

Reply:
🗒️ Decided to keep this here since `product` won't always be passed as a prop.

Reply:
Hopefully we can get rid of it in the process of refactors that allow us to not have to manually hydrate `CurrentProductContext` anymore. 😊